docs/tables/aws_s3_server_access_log/queries.md

```diff
@@ -0,0 +1,272 @@
+## Access Log Examples
+
+### Daily access trends
+
+Count access log entries per day to identify trends over time.
+
+```sql
+select
+  strftime(timestamp, '%y-%m-%d') AS access_date,
+  count(*) AS access_count
+from
+  aws_s3_server_access_log
+group by
+  access_date
+order by
+  access_date asc;
+```
+
+### Top 10 accessed objects
+
+List the 10 most frequently accessed S3 objects.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  key,
+  count(*) as access_count
+from
+  aws_s3_server_access_log
+group by
+  timestamp,
+  requester,
+  bucket,
+  key
+order by
+  access_count desc
+limit 10;
+```
+
+### Top 10 requesters
+
+Identify the top 10 requesters generating the most traffic.
+
+```sql
+select
+  timestamp,
+  requester,
+  count(*) as request_count
+from
+  aws_s3_server_access_log
+group by
+  timestamp,
+  requester
+order by
+  request_count desc
+limit 10;
+```
+
+### Top error codes
+
+Identify the most frequent error codes.
+
+```sql
+select
+  timestamp,
+  requester,
+  error_code,
+  count(*) as access_count
+from
+  aws_s3_server_access_log
+where
+  error_code is not null
+group by
+  timestamp,
+  requester,
+  error_code
+order by
+  access_count desc;
+```
+
+## Detection Examples
+
+### Unusual large file downloads
+
+Detect unusually large downloads from S3.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  key,
+  bytes_sent,
+  remote_ip,
+  user_agent
+from
+  aws_s3_server_access_log
+where
+  bytes_sent is not null
+  and bytes_sent > 50000000 -- 50MB
+order by
+  bytes_sent desc;
+```
+
+### Requests from unknown IPs
+
+Identify S3 access from unknown or unapproved IP addresses.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  remote_ip,
+  operation
+from
+  aws_s3_server_access_log
+where
+  remote_ip not in ('192.0.2.146', '206.253.208.100')
+order by
+  timestamp desc;
+```
+
+## Operational Examples
+
+### Find failed requests
+
+Identify all S3 requests that resulted in an error.
+
+```sql
+select
+  timestamp,
+  requester,
+  bucket,
+  request_uri,
+  http_status,
+  error_code,
+  remote_ip,
+  user_agent
+from
+  aws_s3_server_access_log
+where
+  http_status is not null
```

Comment:
Are 300 errors fatal in S3 requests?

Reply:
No, a 300 error code in an S3 request is generally not considered "fatal" as it usually indicates a "redirect" , meaning the request needs to be sent to a different endpoint to reach the desired object
https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html